### ISSUE TITLE
[🔥] Issues Integrating Firebase Analytics into Project

### ISSUE THREAD

**Issue by amit-lekhak** [invertase/react-native-firebase#7604]
## Issue

I have been attempting to integrate Firebase Analytics into an existing project without success. To troubleshoot, I created a new project using the latest packages and followed all the necessary steps. The app builds successfully, but I am encountering several issues:

1. **DebugView:** No logs are appearing in the DebugView, and connected devices are not being displayed.

2. **Debug Mode:** I used terminal commands to enable debug mode for the code, as evidenced by the logs.

3. **Logcat Results:** The logcat results indicate that the service is being connected, tasks are being queued, but the service disconnects due to inactivity. Additionally, the gms log shows a SQLite error: `SQLiteConstraintException: UNIQUE constraint failed: pending_ops.tag` when attempting to log an event manually.

4. **iOS:** On iOS, events appear to be logged in the xcode console, but the log shows a network error warning: `Could not connect to the server.`

```csharp
2024-02-01 12:53:44.716 16720-16720 FirebaseApp             com.demoanalytics                    I  Device unlocked: initializing all Firebase APIs for app [DEFAULT]
2024-02-01 12:53:44.884 16720-16749 FA                      com.demoanalytics                    V  App measurement collection enabled
2024-02-01 12:53:44.885 16720-16749 FA                      com.demoanalytics                    V  App measurement enabled for app package, google app id: com.demoanalytics, 1:38381271549:android:3e0d68c923aba3e5580194
2024-02-01 12:53:44.886 16720-16749 FA                      com.demoanalytics                    I  App measurement initialized, version: 82001
2024-02-01 12:53:44.886 16720-16749 FA                      com.demoanalytics                    I  To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
2024-02-01 12:53:44.886 16720-16749 FA                      com.demoanalytics                    I  Faster debug mode event logging enabled. To disable, run:
                                                                                                      adb shell setprop debug.firebase.analytics.app .none.
2024-02-01 12:53:44.886 16720-16749 FA                      com.demoanalytics                    D  Debug-level message logging enabled
2024-02-01 12:53:44.920 16720-16749 FA                      com.demoanalytics                    D  Resetting session stitching token to not null
2024-02-01 12:53:44.977 16720-16749 FA                      com.demoanalytics                    V  Connecting to remote service
2024-02-01 12:53:45.015 16720-16749 FA                      com.demoanalytics                    V  Connection attempt already in progress
2024-02-01 12:53:45.018 16720-16749 FA                      com.demoanalytics                    V  Connection attempt already in progress
2024-02-01 12:53:45.035 16720-16745 FA                      com.demoanalytics                    V  onActivityCreated
2024-02-01 12:53:45.083 16720-16749 FA                      com.demoanalytics                    V  Activity resumed, time: 301814197
2024-02-01 12:53:45.095 16720-16749 FA                      com.demoanalytics                    I  Tag Manager is not found and thus will not be used
2024-02-01 12:53:45.155 16720-16749 FA                      com.demoanalytics                    V  Connection attempt already in progress
2024-02-01 12:53:45.156 16720-16749 FA                      com.demoanalytics                    V  Connection attempt already in progress
2024-02-01 12:53:45.280 16720-16749 FA                      com.demoanalytics                    D  Connected to remote service
2024-02-01 12:53:45.280 16720-16749 FA                      com.demoanalytics                    V  Processing queued up service tasks: 5
2024-02-01 12:53:50.658 16720-16749 FA                      com.demoanalytics                    V  Inactivity, disconnecting from the service
2024-02-01 12:54:18.280 16720-16749 FA                      com.demoanalytics                    V  Connecting to remote service
2024-02-01 12:54:18.322 16720-16749 FA                      com.demoanalytics                    D  Connected to remote service
2024-02-01 12:54:18.322 16720-16749 FA                      com.demoanalytics                    V  Processing queued up service tasks: 1
2024-02-01 12:54:23.347 16720-16749 FA                      com.demoanalytics                    V  Inactivity, disconnecting from the service

```

```csharp
2024-02-01 13:01:54.595 11056-17321 TextClassifierService   com...gle.android.inputmethod.latin  D  No configured system TextClassifierService
2024-02-01 13:01:54.743 16720-17324 FA                      com.demoanalytics                    V  Connecting to remote service
2024-02-01 13:01:54.763 15043-15043 BoundBrokerSvc          com.google.android.gms               D  onBind: Intent { act=com.google.android.gms.measurement.START pkg=com.google.android.gms }
2024-02-01 13:01:54.763 15043-15043 BoundBrokerSvc          com.google.android.gms               D  Loading bound service for intent: Intent { act=com.google.android.gms.measurement.START pkg=com.google.android.gms }
2024-02-01 13:01:54.777 16720-17324 FA                      com.demoanalytics                    D  Connected to remote service
2024-02-01 13:01:54.777 16720-17324 FA                      com.demoanalytics                    V  Processing queued up service tasks: 1
2024-02-01 13:01:54.836 11555-11555 BoundBrokerSvc          com.google.android.gms.persistent    D  onBind: Intent { act=com.google.android.gms.presencemanager.service.START dat=chimera-action: cmp=com.google.android.gms/.chimera.PersistentApiService }
2024-02-01 13:01:54.836 11555-11555 BoundBrokerSvc          com.google.android.gms.persistent    D  Loading bound service for intent: Intent { act=com.google.android.gms.presencemanager.service.START dat=chimera-action: cmp=com.google.android.gms/.chimera.PersistentApiService }
2024-02-01 13:01:54.841 11555-11555 BoundBrokerSvc          com.google.android.gms.persistent    D  onBind: Intent { act=com.google.android.gms.presencemanager.service.INTERNAL_IDENTITY dat=chimera-action: cmp=com.google.android.gms/.chimera.PersistentApiService }
2024-02-01 13:01:54.841 11555-11555 BoundBrokerSvc          com.google.android.gms.persistent    D  Loading bound service for intent: Intent { act=com.google.android.gms.presencemanager.service.INTERNAL_IDENTITY dat=chimera-action: cmp=com.google.android.gms/.chimera.PersistentApiService }
2024-02-01 13:01:54.862 11555-17328 NetworkScheduler        com.google.android.gms.persistent    W  Error inserting period=2000 preferred_charging_state=1 preferred_network_type=1 target_class=com.google.android.gms.measurement.PackageMeasurementTaskService required_network_type=0 runtime=1706771814859 required_idleness_state=0 source=16 service_kind=0 target_package=com.google.android.gms persistence_level=1 source_version=235014000 last_runtime=0 user_id=0 job_id=-1 requires_charging=0 tag=Measurement.PackageMeasurementTaskService.UPLOAD_TASK_TAG flex_time=0 task_type=0 retry_strategy={"maximum_backoff_seconds":{"3600":0},"initial_backoff_seconds":{"30":0},"retry_policy":{"0":0}} [CONTEXT service_id=218 ]
                                                                                                    android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: pending_ops.tag, pending_ops.target_class, pending_ops.target_package, pending_ops.user_id (code 2067 SQLITE_CONSTRAINT_UNIQUE)
   at android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId(Native Method)
   at android.database.sqlite.SQLiteConnection.executeForLastInsertedRowId(SQLiteConnection.java:796 undefined)
   at android.database.sqlite.SQLiteSession.executeForLastInsertedRowId(SQLiteSession.java:788 undefined)
   at android.database.sqlite.SQLiteStatement.executeInsert(SQLiteStatement.java:86 undefined)
   at android.database.sqlite.SQLiteDatabase.insertWithOnConflict(SQLiteDatabase.java:1564 undefined)
   at android.database.sqlite.SQLiteDatabase.insertOrThrow(SQLiteDatabase.java:1459 undefined)
   at cesi.d(:com.google.android.gms@235014022@23.50.14 (100400-592568214):719)
   at ceqm.n(:com.google.android.gms@235014022@23.50.14 (100400-592568214):29)
   at ceqm.u(:com.google.android.gms@235014022@23.50.14 (100400-592568214):362)
   at ceqm.h(:com.google.android.gms@235014022@23.50.14 (100400-592568214):51)
   at celk.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):72)
   at almc.c(:com.google.android.gms@235014022@23.50.14 (100400-592568214):50)
   at almc.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):76)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167 undefined)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641 undefined)
   at alrs.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):8)
   at java.lang.Thread.run(Thread.java:764 undefined)
2024-02-01 13:01:55.150  3247-3572  GnssLocationProvider    system_server                        I  WakeLock acquired by sendMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@3d10e1d)
2024-02-01 13:01:55.150  3247-3260  GnssLocationProvider    system_server                        I  WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@3d10e1d)
2024-02-01 13:01:55.385 11555-16780 NetworkScheduler        com.google.android.gms.persistent    W  Error inserting period=3602000 preferred_charging_state=1 preferred_network_type=1 target_class=com.google.android.gms.measurement.PackageMeasurementTaskService required_network_type=0 runtime=1706771815383 required_idleness_state=0 source=16 service_kind=0 target_package=com.google.android.gms persistence_level=1 source_version=235014000 last_runtime=0 user_id=0 job_id=-1 requires_charging=0 tag=Measurement.PackageMeasurementTaskService.UPLOAD_TASK_TAG flex_time=1800000 task_type=0 retry_strategy={"maximum_backoff_seconds":{"3600":0},"initial_backoff_seconds":{"30":0},"retry_policy":{"0":0}} [CONTEXT service_id=218 ]
                                                                                                    android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: pending_ops.tag, pending_ops.target_class, pending_ops.target_package, pending_ops.user_id (code 2067 SQLITE_CONSTRAINT_UNIQUE)
   at android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId(Native Method)
   at android.database.sqlite.SQLiteConnection.executeForLastInsertedRowId(SQLiteConnection.java:796 undefined)
   at android.database.sqlite.SQLiteSession.executeForLastInsertedRowId(SQLiteSession.java:788 undefined)
   at android.database.sqlite.SQLiteStatement.executeInsert(SQLiteStatement.java:86 undefined)
   at android.database.sqlite.SQLiteDatabase.insertWithOnConflict(SQLiteDatabase.java:1564 undefined)
   at android.database.sqlite.SQLiteDatabase.insertOrThrow(SQLiteDatabase.java:1459 undefined)
   at cesi.d(:com.google.android.gms@235014022@23.50.14 (100400-592568214):719)
   at ceqm.n(:com.google.android.gms@235014022@23.50.14 (100400-592568214):29)
   at ceqm.u(:com.google.android.gms@235014022@23.50.14 (100400-592568214):362)
   at ceqm.h(:com.google.android.gms@235014022@23.50.14 (100400-592568214):51)
   at celk.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):72)
   at almc.c(:com.google.android.gms@235014022@23.50.14 (100400-592568214):50)
   at almc.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):76)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167 undefined)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641 undefined)
   at alrs.run(:com.google.android.gms@235014022@23.50.14 (100400-592568214):8)
   at java.lang.Thread.run(Thread.java:764 undefined)
2024-02-01 13:01:56.153  3247-3572  GnssLocationProvider    system_server                        I  WakeLock acquired by sendMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@9be0892)
2024-02-01 13:01:56.153  3247-3260  GnssLocationProvider    system_server                        I  WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@9be0892)
2024-02-01 13:01:57.154  3247-3572  GnssLocationProvider    system_server                        I  WakeLock acquired by sendMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@f725263)
2024-02-01 13:01:57.154  3247-3260  GnssLocationProvider    system_server                        I  WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@f725263)
2024-02-01 13:01:58.160  3247-3572  GnssLocationProvider    system_server                        I  WakeLock acquired by sendMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@4596e60)
2024-02-01 13:01:58.160  3247-3260  GnssLocationProvider    system_server                        I  WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@4596e60)
2024-02-01 13:01:59.161  3247-3572  GnssLocationProvider    system_server                        I  WakeLock acquired by sendMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@7f56f19)
2024-02-01 13:01:59.162  3247-3260  GnssLocationProvider    system_server                        I  WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@7f56f19)
2024-02-01 13:01:59.797 16720-17324 FA                      com.demoanalytics                    V  Inactivity, disconnecting from the service
```
IOS logs

```csharp
10.20.0 - [FirebaseCore][I-COR000001] Configuring the default app.
10.20.0 - [FirebaseCore][I-COR000033] Data Collection flag is not set.
<SKPaymentQueue: 0x6000013887b0>: No observers found that respond to "paymentQueue:shouldAddStorePayment:forProduct:", will not check for purchase intents
10.20.0 - [FirebaseAnalytics][I-ACS024000] Debug mode is on
10.20.0 - [FirebaseInstallations][I-FIS002000] -[FIRInstallationsIDController createGetInstallationItemPromise], appName: __FIRAPP_DEFAULT
10.20.0 - [GULReachability][I-REA902003] Monitoring the network status
10.20.0 - [GULReachability][I-REA902003] Monitoring the network status
10.20.0 - [FirebaseAnalytics][I-ACS023007] Analytics v.10.20.0 started
10.20.0 - [FirebaseAnalytics][I-ACS023008] To disable debug logging set the following application argument: -noFIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
10.20.0 - [FirebaseAnalytics][I-ACS023009] Debug logging enabled
10.20.0 - [FirebaseAnalytics][I-ACS023207] To enable verbose logging set the following application argument: -FIRAnalyticsVerboseLoggingEnabled (see http://goo.gl/RfcP7r)
10.20.0 - [FirebaseAnalytics][I-ACS044000] GoogleAppMeasurementIdentitySupport dependency is linked.
10.20.0 - [FirebaseAnalytics][I-ACS044002] The AdSupport Framework is not currently linked. Some features will not function properly. Learn more at http://goo.gl/9vSsPb
10.20.0 - [FirebaseAnalytics][I-ACS002002] APMExperimentAlarm scheduled to fire in approx. (s): 9.999958992004395
RCTBridge required dispatch_sync to load RCTAccessibilityManager. This may lead to deadlocks
Running application demoAnalytics ({
    initialProps =     {
    };
    rootTag = 1;
})
10.20.0 - [GULReachability][I-REA902004] Network status has changed. Code:2, status:Connected
10.20.0 - [GULReachability][I-REA902004] Network status has changed. Code:2, status:Connected
10.20.0 - [GoogleUtilities/AppDelegateSwizzler][I-SWZ001008] Successfully created App Delegate Proxy automatically. To disable the proxy, set the flag GoogleUtilitiesAppDelegateProxyEnabled to NO (Boolean) in the Info.plist
+[RNFBAppModule getCustomDomain:] [Line 239] authDomains:
10.20.0 - [FirebaseAnalytics][I-ACS031010] Tracking view controller. Class, ID: UIViewController, 5869943052233933468
10.20.0 - [FirebaseInstallations][I-FIS002001] -[FIRInstallationsIDController installationWithValidAuthTokenForcingRefresh:0], appName: __FIRAPP_DEFAULT
10.20.0 - [FirebaseAnalytics][I-ACS800023] No pending snapshot to activate. SDK name: app_measurement
Thread Performance Checker: Thread running at User-interactive quality-of-service class waiting on a lower QoS thread running at Default quality-of-service class. Investigate ways to avoid priority inversions
PID: 7903, TID: 65328
…
10.20.0 - [FirebaseAnalytics][I-ACS029017] Configuration was not found in database. The configuration will be fetched from the network when necessary
10.20.0 - [FirebaseAnalytics][I-ACS023016] Analytics is ready to receive events
10.20.0 - [FirebaseAnalytics][I-ACS031006] View controller already tracked. Class, ID: UIViewController, 5869943052233933468
10.20.0 - [FirebaseAnalytics][I-ACS023087] User property set. Name, value: ga_session_id (_sid), 1706766121
10.20.0 - [FirebaseAnalytics][I-ACS023087] User property set. Name, value: ga_session_number (_sno), 2
10.20.0 - [FirebaseAnalytics][I-ACS023051] Logging event: origin, name, params: auto, session_start (_s), {
    ga_event_origin (_o) = auto;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
    session_id (_sid) = 1706766121;
    session_number (_sno) = 2;
}
10.20.0 - [FirebaseAnalytics][I-ACS023073] Debug mode is enabled. Marking event as debug and real-time. Event name, parameters: session_start (_s), {
    ga_debug (_dbg) = 1;
    ga_event_origin (_o) = auto;
    ga_realtime (_r) = 1;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
    session_id (_sid) = 1706766121;
    session_number (_sno) = 2;
}
10.20.0 - [FirebaseAnalytics][I-ACS023072] Event logged. Event name, event params: session_start (_s), {
    ga_debug (_dbg) = 1;
    ga_event_origin (_o) = auto;
    ga_realtime (_r) = 1;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
    session_id (_sid) = 1706766121;
    session_number (_sno) = 2;
}
10.20.0 - [FirebaseAnalytics][I-ACS002002] Measurement timer scheduled to fire in approx. (s): -0.3909770250320435
10.20.0 - [FirebaseAnalytics][I-ACS032014] AdServices framework is not linked. Search Ad Attribution Reporter is disabled.
10.20.0 - [FirebaseAnalytics][I-ACS002003] Measurement timer canceled
10.20.0 - [FirebaseAnalytics][I-ACS002002] Measurement timer scheduled to fire in approx. (s): -0.3921569585800171
10.20.0 - [FirebaseAnalytics][I-ACS002002] Engagement timer scheduled to fire in approx. (s): 3600
10.20.0 - [FirebaseAnalytics][I-ACS023012] Analytics collection enabled
10.20.0 - [FirebaseAnalytics][I-ACS023220] Analytics screen reporting is enabled. Call Analytics.logEvent(AnalyticsEventScreenView, parameters: [...]) to log a screen view event. To disable automatic screen reporting, set the flag FirebaseAutomaticScreenReportingEnabled to NO (boolean) in the Info.plist
10.20.0 - [FirebaseAnalytics][I-ACS023051] Logging event: origin, name, params: auto, screen_view (_vs), {
    ga_event_origin (_o) = auto;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
}
10.20.0 - [FirebaseAnalytics][I-ACS023073] Debug mode is enabled. Marking event as debug and real-time. Event name, parameters: screen_view (_vs), {
    ga_debug (_dbg) = 1;
    ga_event_origin (_o) = auto;
    ga_realtime (_r) = 1;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
}
10.20.0 - [FirebaseAnalytics][I-ACS023072] Event logged. Event name, event params: screen_view (_vs), {
    ga_debug (_dbg) = 1;
    ga_event_origin (_o) = auto;
    ga_realtime (_r) = 1;
    ga_screen_class (_sc) = UIViewController;
    ga_screen_id (_si) = 5869943052233933468;
}
10.20.0 - [FirebaseAnalytics][I-ACS002001] Measurement timer fired
10.20.0 - [FirebaseAnalytics][I-ACS002003] Measurement timer canceled
10.20.0 - [FirebaseAnalytics][I-ACS900001] Downloading data. Host: https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios
nw_socket_handle_socket_event [C5.1.1.1:3] Socket SO_ERROR 61
nw_socket_handle_socket_event [C5.1.1.2:3] Socket SO_ERROR 61
Connection 5: received failure notification
Connection 5: failed to connect 1:61, reason -1
Connection 5: encountered error(1:61)
Task <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1> HTTP load failed, 0/0 bytes (error code: -1,004 [1:61])
Task <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1> finished with error [-1,004] Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001f090e0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, NSErrorFailingURLKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS901017] Encounter network error. Code, error: -1004, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001f090e0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, NSErrorFailingURLKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS023128] Network request failed to fetch remote config. Code, Error: 0, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001f090e0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, NSErrorFailingURLKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS023129] Network fetch failed. Will retry later. Code, error: 0, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001f090e0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <4113DC3B-D6A4-4EE3-B6EF-13CC4B0CCF3C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, NSErrorFailingURLKey=https://app-measurement.com/config/app/1:38381271549:ios:e1964f7a076a8e11580194?app_instance_id=83F700F137424BAD941373AE58A3DFF1&gmp_version=102000&runtime_version=0&platform=ios, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS002002] Measurement timer scheduled to fire in approx. (s): 1814.36427295208
10.20.0 - [FirebaseAnalytics][I-ACS002001] APMExperimentAlarm fired
10.20.0 - [FirebaseAnalytics][I-ACS002003] APMExperimentAlarm canceled
nw_socket_handle_socket_event [C6.1.1.1:3] Socket SO_ERROR 61
nw_socket_handle_socket_event [C6.1.1.2:3] Socket SO_ERROR 61
Connection 6: received failure notification
Connection 6: failed to connect 1:61, reason -1
Connection 6: encountered error(1:61)
Task <368366FC-BB43-4528-8507-F3CBD062591C>.<1> HTTP load failed, 0/0 bytes (error code: -1,004 [1:61])
Task <368366FC-BB43-4528-8507-F3CBD062591C>.<1> finished with error [-1,004] Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001fe1da0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/sdk-exp, NSErrorFailingURLKey=https://app-measurement.com/sdk-exp, _kCFStreamErrorDomainKey=1}
<Google/Utilities/Network/ERROR> Encounter network error. Code, error: -1004, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001fe1da0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/sdk-exp, NSErrorFailingURLKey=https://app-measurement.com/sdk-exp, _kCFStreamErrorDomainKey=1}
10.20.0 - [GULNetwork][I-NET901017] <Google/Utilities/Network/ERROR> Encounter network error. Code, error: -1004, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001fe1da0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/sdk-exp, NSErrorFailingURLKey=https://app-measurement.com/sdk-exp, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS820002] Network fetch failed. Will retry later. Code, error: 0, Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600001fe1da0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalUploadTask <368366FC-BB43-4528-8507-F3CBD062591C>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=https://app-measurement.com/sdk-exp, NSErrorFailingURLKey=https://app-measurement.com/sdk-exp, _kCFStreamErrorDomainKey=1}
10.20.0 - [FirebaseAnalytics][I-ACS002002] APMExperimentAlarm scheduled to fire in approx. (s): 1789.59762597084
nw_socket_handle_socket_event [C7.1.1:1] Socket SO_ERROR 61
Unbalanced calls start/end for tag 19
nw_socket_handle_socket_event [C7.1.2:1] Socket SO_ERROR 61
nw_connection_get_connected_socket_block_invoke [C7] Client called nw_connection_get_connected_socket on unconnected nw_connection
TCP Conn 0x600002081c20 Failed : error 0:61 [61]
RCTBridge required dispatch_sync to load RCTAccessibilityManager. This may lead to deadlocks
Running "demoAnalytics" with {"rootTag":1,"initialProps":{}}
10.20.0 - [FirebaseAnalytics][I-ACS031006] View controller already tracked. Class, ID: UIViewController, 5869943052233933468
```

[![](https://img.shields.io/badge/Formatted%20by-https%3A%2F%2Fst.elmah.io-%230da58e)](https://st.elmah.io)

The repo for said project is at: [Demo Analytics](https://github.com/amit-lekhak/demoAnalytics)

#### I have carefully followed the steps outlined in the documentation, but I am uncertain if I missed any crucial details. Any assistance in resolving these issues would be greatly appreciated. Thank you.
---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
  "name": "demoAnalytics",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@react-native-firebase/analytics": "^18.8.0",
    "@react-native-firebase/app": "^18.8.0",
    "react": "18.2.0",
    "react-native": "0.73.3"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/babel-preset": "0.73.20",
    "@react-native/eslint-config": "0.73.2",
    "@react-native/metro-config": "0.73.4",
    "@react-native/typescript-config": "0.73.1",
    "@types/react": "^18.2.6",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.6.3",
    "eslint": "^8.19.0",
    "jest": "^29.6.3",
    "prettier": "2.8.8",
    "react-test-renderer": "18.2.0",
    "typescript": "5.0.4"
  },
  "engines": {
    "node": ">=18"
  }
}

```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# Resolve react_native_pods.rb with node to allow for hoisting
require Pod::Executable.execute_command('node', ['-p',
  'require.resolve(
    "react-native/scripts/react_native_pods.rb",
    {paths: [process.argv[1]]},
  )', __dir__]).strip

platform :ios, min_ios_version_supported
prepare_react_native_project!

# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
# ```js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
# ```
flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target 'demoAnalytics' do
  config = use_native_modules!
  use_frameworks! :linkage => :static
  $RNFirebaseAsStaticFramework = true
  
  use_react_native!(
    :path => config[:reactNativePath],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
#    :flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'demoAnalyticsTests' do
    inherit! :complete
    # Pods for testing
  end

  post_install do |installer|
    # https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/react_native_pods.rb#L197-L202
    react_native_post_install(
      installer,
      config[:reactNativePath],
      :mac_catalyst_enabled => false
    )
  end
end

```

#### `AppDelegate.m`:

```objc
#import "AppDelegate.h"
#import <Firebase.h>

#import <React/RCTBundleURLProvider.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  [FIRApp configure];

  self.moduleName = @"demoAnalytics";
  // You can add your custom initial props in the dictionary below.
  // They will be passed down to the ViewController used by React Native.
  self.initialProps = @{};

  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
  return [self getBundleURL];
}

- (NSURL *)getBundleURL
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

@end

```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
buildscript {
    ext {
        buildToolsVersion = "34.0.0"
        minSdkVersion = 21
        compileSdkVersion = 34
        targetSdkVersion = 34
        ndkVersion = "25.1.8937393"
        kotlinVersion = "1.8.0"
    }
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle")
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin")
        classpath('com.google.gms:google-services:4.4.0')

    }
}

apply plugin: "com.facebook.react.rootproject"

```

#### `android/app/build.gradle`:

```groovy
apply plugin: "com.android.application"
apply plugin: "org.jetbrains.kotlin.android"
apply plugin: "com.facebook.react"

/**
 * This is the configuration block to customize your React Native Android app.
 * By default you don't need to apply any configuration, just uncomment the lines you need.
 */
react {
    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    //   The folder where the react-native Codegen package is. Default is ../node_modules/@react-native/codegen
    // codegenDir = file("../node_modules/@react-native/codegen")
    //   The cli.js file which is the React Native CLI entrypoint. Default is ../node_modules/react-native/cli.js
    // cliFile = file("../node_modules/react-native/cli.js")

    /* Variants */
    //   The list of variants to that are debuggable. For those we're going to
    //   skip the bundling of the JS bundle and the assets. By default is just 'debug'.
    //   If you add flavors like lite, prod, etc. you'll have to list your debuggableVariants.
    // debuggableVariants = ["liteDebug", "prodDebug"]

    /* Bundling */
    //   A list containing the node command and its flags. Default is just 'node'.
    // nodeExecutableAndArgs = ["node"]
    //
    //   The command to run when bundling. By default is 'bundle'
    // bundleCommand = "ram-bundle"
    //
    //   The path to the CLI configuration file. Default is empty.
    // bundleConfig = file(../rn-cli.config.js)
    //
    //   The name of the generated asset file containing your JS bundle
    // bundleAssetName = "MyApplication.android.bundle"
    //
    //   The entry file for bundle generation. Default is 'index.android.js' or 'index.js'
    // entryFile = file("../js/MyApplication.android.js")
    //
    //   A list of extra flags to pass to the 'bundle' commands.
    //   See https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle
    // extraPackagerArgs = []

    /* Hermes Commands */
    //   The hermes compiler command to run. By default it is 'hermesc'
    // hermesCommand = "$rootDir/my-custom-hermesc/bin/hermesc"
    //
    //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
    // hermesFlags = ["-O", "-output-source-map"]
}

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US. Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

android {
    ndkVersion rootProject.ext.ndkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion
    compileSdk rootProject.ext.compileSdkVersion

    namespace "com.demoanalytics"
    defaultConfig {
        applicationId "com.demoanalytics"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
    }
    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
}

dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    implementation("com.facebook.react:flipper-integration")

    if (hermesEnabled.toBoolean()) {
        implementation("com.facebook.react:hermes-android")
    } else {
        implementation jscFlavor
    }
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
apply plugin: 'com.google.gms.google-services' // <- Add this line

```

#### `android/settings.gradle`:

```groovy
rootProject.name = 'demoAnalytics'
apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
include ':app'
includeBuild('../node_modules/@react-native/gradle-plugin')

```

#### `MainApplication.kt`:

```java
package com.demoanalytics

import android.app.Application
import com.facebook.react.PackageList
import com.facebook.react.ReactApplication
import com.facebook.react.ReactHost
import com.facebook.react.ReactNativeHost
import com.facebook.react.ReactPackage
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
import com.facebook.react.defaults.DefaultReactNativeHost
import com.facebook.react.flipper.ReactNativeFlipper
import com.facebook.soloader.SoLoader

class MainApplication : Application(), ReactApplication {

  override val reactNativeHost: ReactNativeHost =
      object : DefaultReactNativeHost(this) {
        override fun getPackages(): List<ReactPackage> =
            PackageList(this).packages.apply {
              // Packages that cannot be autolinked yet can be added manually here, for example:
              // add(MyReactNativePackage())
            }

        override fun getJSMainModuleName(): String = "index"

        override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG

        override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
        override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
      }

  override val reactHost: ReactHost
    get() = getDefaultReactHost(this.applicationContext, reactNativeHost)

  override fun onCreate() {
    super.onCreate()
    SoLoader.init(this, false)
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      // If you opted-in for the New Architecture, we load the native entry point for this app.
      load()
    }
    ReactNativeFlipper.initializeFlipper(this, reactNativeHost.reactInstanceManager)
  }
}

```

#### `AndroidManifest.xml`:

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android">

    <uses-permission android:name="android.permission.INTERNET" />

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize|uiMode"
        android:launchMode="singleTask"
        android:windowSoftInputMode="adjustResize"
        android:exported="true">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
    </application>
</manifest>

```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
  OS: macOS 13.6.1
  CPU: (8) arm64 Apple M1
  Memory: 87.30 MB / 8.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.0
    path: ~/.nvm/versions/node/v20.11.0/bin/node
  Yarn:
    version: 1.22.17
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.0/bin/npm
  Watchman:
    version: 2023.11.13.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /Users/coresoftmac2022/.rvm/gems/ruby-2.6.10/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK:
    API Levels:
      - "28"
      - "30"
      - "31"
      - "32"
      - "33"
      - "34"
    Build Tools:
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-28 | Google ARM64-V8a Play ARM 64 v8a
      - android-30 | Intel x86 Atom_64
      - android-30 | Google APIs Intel x86 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 15.0.1/15A507
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /Users/coresoftmac2022/.rvm/rubies/ruby-2.6.10/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.3
    wanted: 0.73.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [x] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `18.8.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `analytics`
- **Are you using `TypeScript`?**
  - `Y` & `5.0.4`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
This sounds like either your specific devices or something on your network is blocking the google analytics network path, likely for privacy reasons. It's quite common actually - every few months someone does an otherwise-perfect integration but with no analytics showing up where they seem them and the root cause is a network block against google analytics.

I'd triple-check that, perhaps try different network path (VPN or something), etc

**Comment by amit-lekhak** [comment]
Thanks for the insight! You were spot on. It seems there is indeed some network issue blocking the Google Analytics path. Using a VPN did the trick for now, and for Android, emulator device also needed an update. Appreciate your help in troubleshooting this! 👍 😀

**Comment by mikehardy** [comment]
glad that worked!

**Comment by AuroPick** [comment]
I was getting the same error. Using vpn or switching to cellular fixed my problem but I don't know how to fix my wifi.
